src/eval-frame/components/reps/value-renderer.jsx

```diff
@@ -4,7 +4,25 @@ import PropTypes from "prop-types";
 import DefaultRenderer from "./default-handler";
 import ErrorRenderer from "./error-handler";
 import HTMLHandler from "./html-handler";
+import TableRenderer from "./data-table-rep";
 import UserReps from "./user-reps-manager";
+import { isRowDf } from "./rep-utils/rep-type-chooser";
+
+function repChooser(value) {
+  if (UserReps.getUserRepIfAvailable(value)) {
+    return "GLOBAL_USER_RENDERER";
+  }
+  if (value && value.iodideRender instanceof Function) {
```

Comment:
what's the diff between this and the `GLOBAL_USER_RENDERER`? Is there a reason we're supporting both with two different bits of functionality? I guess one might be provided by a language, the other not. If that's the case, it'd be great to have flags that are a bit closer to what the use case actually is.

Reply:
the difference between the GLOBAL_USER_RENDERER and OBJECT_USER_RENDER_METHOD is that GLOBAL_USER_RENDERER "globally" checks all values that are returned from an eval to see whether they match the criteria for the user rendered, whereas the other case uses a render method in the `iodideRender` property.

the former is set using `iodide.addOutputRenderer(rendererSpecification)` (see https://iodide-project.github.io/docs/api/ ); the latter is a method manually attached to any object.

these were distinct use cases that @mdboom envisioned, and they seem reasonable; wrt to the terminology, i'm certainly open to suggestions. do either of you two have ideas?

Reply:
got it, that makes a lot of sense, and I like that distinction (which is obvious now from the code, just didn't realize we had `.iodideRender` :) ). We want a way for users to dynamically render content that doesn't involve adding an output renderer, which I'm sure is useful for all sorts of things including language plugins. I think I was thrown off by the term `GLOBAL_USER_RENDERER`, since in my head it's more like `USER_REGISTERED_RENDERER` or `CUSTOM_REGISTERED_RENDERER` or something like that, since it suggests that the thing was added to the rep checker

Reply:
changed to "USER_REGISTERED_RENDERER"